test/dlc_tests/sanity/test_dlc_labels.py

```diff
@@ -42,4 +44,71 @@ def test_dlc_major_version_label(image, region):
 
     assert major_version, f"{image} has no LABEL named 'dlc_major_version'. Please insert label."
 
-    LOGGER.info(f"{image} has 'dlc_major_version' = {major_version}")
+    test_utils.LOGGER.info(f"{image} has 'dlc_major_version' = {major_version}")
+
+
+@pytest.mark.integration("dlc_major_version_label")
+@pytest.mark.model("N/A")
+def test_dlc_version_dockerfiles(image):
+    """
+    Test to make sure semantic versioning scheme in Dockerfiles is correct
+
+    :param image: <str> ECR image URI
+    """
+    dlc_dir = os.getcwd().split(f"{os.sep}test{os.sep}")[0]
+    job_type = test_utils.get_job_type_from_image(image)
+    framework, fw_version = test_utils.get_framework_and_version_from_tag(image)
+    processor = test_utils.get_processor_from_image_uri(image)
+
+    root_dir = os.path.join(dlc_dir, framework, job_type, "docker")
+
+    # Skip older FW versions that did not use this versioning scheme
+    references = {"tensorflow2": "2.2.0", "tensorflow1": "1.16.0", "mxnet": "1.7.0", "pytorch": "1.5.0"}
+    if test_utils.is_tf1(image):
+        reference_fw = "tensorflow1"
+    elif test_utils.is_tf2(image):
+        reference_fw = "tensorflow2"
+    else:
+        reference_fw = framework
+    if processor != "eia" and fw_version < references[reference_fw]:
+        pytest.skip(
+            f"Not enforcing new versioning scheme on old image {image}. "
+            f"Started enforcing version scheme on the following: {references}"
+        )
+
+    dockerfiles = []
+    fw_version_major_minor = re.match(r"(\d+.\d+)", fw_version).group(1)
+    for root, dirnames, filenames in os.walk(root_dir):
+        for filename in filenames:
+            if filename == f"Dockerfile.{processor}":
+                dockerfile_path = os.path.join(root_dir, root, filename)
+                if "example" not in dockerfile_path and f"{os.sep}{fw_version_major_minor}" in dockerfile_path:
+                    dockerfiles.append(dockerfile_path)
```

Comment:
This loop doesn't deal with multiple python versions.

If I were to create a Python 3.6 version in parallel with a Python 3.7 version of a framework version's CPU and GPU dockerfiles, this loop would catch both versions.

It looks like the logic further below will fail, in that case.

Reply:
It should catch both versions, right? Would our version scheme not expect two different DLC major versions for different python versions?